### PR TITLE
Delete cardset

### PR DESCRIPTION
**Pull request by YuzhongHuang**

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -218,12 +218,23 @@ bingo.controller('homeController', function($scope, $http, $location, bingosocke
...
-    //$http.post(...)
-    console.log('Go write a "delete card set" controller');
+    $http.post('/api/delete/cardset', {
+        cardset_id: cardsetid
+      })
+      .success(function(data) {
+        if (data.restrict) {
+          confirm('Sorry but you are not permitted to do that');
+        } else {
+          $window.location.reload();
```
Comment (ryaneggert) [inline public/js/main.js]: Do we have to reload? Surely we could send back a list of all cardsets and update `$scope.cardsets`. 
We may even be able to get by with, upon confirmation of deletion, remove the deleted cardset from `$scope.cardsets` 

[ Note on the second option -- to remove an item from an array, you can probably follow the `.indexOf()` and `.splice()` process as described [here](http://stackoverflow.com/a/5767357). ]

Comment (YuzhongHuang) [inline public/js/main.js]: OK, Ryan

I will do it~

File: server.js [hunk 2]
```diff
@@ -44,8 +44,18 @@ app.post('/api/game/initialize', auths.isAuth_api, game.init);
 
 app.get('/api/user/cardsets', auths.isAuth_api, game.getUserCardsets);
 
+app.post('/api/delete/cardset', game.deleteCardset);
```
Comment (ryaneggert) [inline server.js]: Could you please add the auths.isAuth_api middleware to your new api calls?

File: public/js/main.js [hunk 3]
```diff
@@ -220,7 +283,7 @@ bingo.controller('addGameController', function($scope, $http, $location) {
   };
 });
 
-bingo.controller('homeController', function($scope, $http, $location, bingosockets) {
+bingo.controller('homeController', function($scope, $http, $location, $window, bingosockets) {
```
Comment (ryaneggert) [inline public/js/main.js]: Do we still need $window?

**Pull request merged**